Sorting give some json error

Hi,

When we want search for some words like "CALL ga.es.queryNode('{"from":0,"size":50,"query":{"match":{"message":"someren asten"}}}') YIELD node return node" it works.

Now we want to sort the results by sended(timestamp). I have read the documentation of Elasticsearch (https://www.elastic.co/guide/en/elasticsearch/reference/current/search-request-sort.html) and i think this command must be work "CALL ga.es.queryNode('{"from":0,"size":50,"query":{"match":{"message":"someren asten"}},"sort":[{"sended":{"order":"desc"}}]}') YIELD node return node" but i get the following error.

**Neo.ClientError.Procedure.ProcedureCallFailed:**
Failed to invoke procedure `ga.es.queryNode`: Caused by: java.lang.UnsupportedOperationException: JsonNull

We hope that any can help us :)

--- a/src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java
+++ b/src/main/java/com/graphaware/module/es/proc/ElasticSearchProcedures.java
@@ -16,6 +16,7 @@
 
 package com.graphaware.module.es.proc;
 
+import com.graphaware.common.log.LoggerFactory;
 import com.graphaware.module.es.ElasticSearchModule;
 import com.graphaware.module.es.proc.result.JsonSearchResult;
 import com.graphaware.module.es.proc.result.NodeSearchResult;
@@ -25,6 +26,8 @@ import com.graphaware.module.es.search.Searcher;
 import org.neo4j.graphdb.GraphDatabaseService;
 import org.neo4j.graphdb.Node;
 import org.neo4j.graphdb.Relationship;
+import org.neo4j.logging.Log;
+import org.neo4j.logging.Logger;
 import org.neo4j.procedure.Context;
 import org.neo4j.procedure.Name;
 import org.neo4j.procedure.PerformsWrites;
@@ -35,6 +38,8 @@ import java.util.stream.Stream;
 import static com.graphaware.runtime.RuntimeRegistry.getStartedRuntime;
 
 public class ElasticSearchProcedures {
+
+    private static final Log LOG = LoggerFactory.getLogger(ElasticSearchProcedures.class);
 
     @Context
     public GraphDatabaseService database;
@@ -50,9 +55,15 @@ public class ElasticSearchProcedures {
     @Procedure("ga.es.queryNode")
     @PerformsWrites
     public Stream<NodeSearchResult> queryNode(@Name("query") String query) {
-        return getSearcher(database).search(query, Node.class).stream().map(match -> {
-            return new NodeSearchResult(match.getItem(), match.score);
-        });
+        try {
+            return getSearcher(database).search(query, Node.class).stream().map(match -> {
+                return new NodeSearchResult(match.getItem(), match.score);
+            });
+        } catch (Exception e) {
+            LOG.error("", e);
+
+            return Stream.empty();
+        }
     }
 
     @Procedure("ga.es.queryRelationship")

--- a/src/main/java/com/graphaware/module/es/search/SearchMatch.java
+++ b/src/main/java/com/graphaware/module/es/search/SearchMatch.java
@@ -23,9 +23,9 @@ public class SearchMatch<T extends PropertyContainer> {
     public final Double score;
     private T item;
 
-    public SearchMatch(String uuid, double score) {
+    public SearchMatch(String uuid, Object score) {
         this.uuid = uuid;
-        this.score = score;
+        this.score = score instanceof Double ? (Double) score : null;
     }
 
     public T getItem() {

--- a/src/main/java/com/graphaware/module/es/search/Searcher.java
+++ b/src/main/java/com/graphaware/module/es/search/Searcher.java
@@ -131,7 +131,7 @@ public class Searcher {
                 .forEach((hitsArray) -> {
                     for (JsonElement element : hitsArray) {
                         JsonObject obj = (JsonObject) element;
-                        Double score = obj.get("_score").getAsDouble();
+                        Double score = obj.get("_score") != null && !obj.get("_score").toString().equals("null") ? Double.valueOf(obj.get("_score").toString()) : null;
                         String keyValue = obj.get("_id") != null ? obj.get("_id").getAsString() : null;
                         if (keyValue == null) {
                             LOG.warn("No key found in search result: " + obj.getAsString());

